base/locks-mt.jl

```diff
@@ -73,3 +73,81 @@ end
 function islocked(l::SpinLock)
     return (@atomic :monotonic l.owned) != 0
 end
+
+const UV_MUTEX_SIZE = ccall(:jl_sizeof_uv_mutex_t, Cint, ())
+
+"""
+    SystemMutex()
+
+Create a system mutex. This is a reentrant lock that is not aware of the julia scheduler.
```

Comment:
```suggestion
Create a system mutex. This is a non-reentrant lock that is not aware of the julia scheduler.
```

The reentrancy implementation is wrong, and not necessary